app/helpers/otp_helper.rb

```diff
@@ -0,0 +1,53 @@
+module OtpHelper
+  def phone_confirmation_instructions
```

Comment:
+1 to presenters. I would also recommend [replacing conditionals with polymorphism](https://sourcemaking.com/refactoring/replace-conditional-with-polymorphism). Let's say you create an `SmsInstructionsPresenter` and a `VoiceInstructionsPresenter`, you can use some metaprogramming to call the appropriate class based on the delivery method. For example, instead of this:
```ruby
if @delivery_method == 'sms'
  t('instructions.2fa.confirm_code_sms', number: @phone_number)
elsif @delivery_method == 'voice'
   t('instructions.2fa.confirm_code_voice', number: @phone_number)
end
```
you can do this:
```ruby
 "#{@delivery_method}InstructionsPresenter".constantize.new(number: @phone_number).instructions
```
or this, if you weren't using presenters and left the method here:
```ruby
t("instructions.2fa.confirm_code_#{@delivery_method}", number: @phone_number)